### PR TITLE
Add support for connections behind from behind proxy

### PR DESCRIPTION
**Pull request by baunz**
Use the available functionality in hdfs storage implementations for GCP and S3

### CODE REVIEW

**Comment by morazow** [comment]
Hey @baunz,

Thanks for the pull request! I am going to check it soon.

File: doc/user_guide/user_guide.md [hunk 1]
```diff
@@ -389,6 +389,22 @@ These optional parameters only apply to the data export statements.
...
+* ``PROXY_HOST`` - Configures a proxy host that is used to access the bucket if this is required 
+
+* ``PROXY_PORT`` - Configures the port to use for the proxy defined with `PROXY_HOST`
+
+* ``PROXY_USERNAME`` - Configures the username to use for the proxy defined with `PROXY_HOST`
+
+* ``PROXY_PASSWORD`` - Configures the port to use for the proxy defined with `PROXY_HOST`
+
+This is only available for S3 and GCP buckets. For Azure Buckets, the proxy can be configured via JVM
+properties with the `%jvmoption`
```
Comment (morazow) [inline doc/user_guide/user_guide.md:406]: Could you please add an example for Azure?

```
%jvmoption -DHTTPS_PROXY=http://username:password@10.10.1.10:1180
```

Comment (morazow) [inline doc/user_guide/user_guide.md:406]:
```suggestion
properties using the `%jvmoption`  when creating deployment scripts.
```

File: src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala [hunk 2]
```diff
@@ -92,6 +92,25 @@ class StorageProperties(private val properties: Map[String, String], private val
   final def isParquetLowercaseSchema(): Boolean =
     get(PARQUET_LOWERCASE_SCHEMA).fold(true)(_.toBoolean)
 
+  /**
+   *  Return the proxy host to use for the bucket
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala:96]:
```suggestion
   *  Returns the proxy host to use for the bucket.
```

File: src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala [hunk 3]
```diff
@@ -92,6 +92,25 @@ class StorageProperties(private val properties: Map[String, String], private val
...
+   */
+  final def getProxyHost(): Option[String] = get(PROXY_HOST)
+
+  /**
+   *  Return the proxy port to use for the bucket
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala:101]:
```suggestion
   *  Returns the proxy port to use for the bucket.
```

Comment (morazow) [reply src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala:101]: Please change other also.

File: src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala [hunk 4]
```diff
@@ -132,6 +151,18 @@ object StorageProperties extends CommonProperties {
   /** An optional property for setting Parquet export schema with lowercase fields. */
   private[storage] final val PARQUET_LOWERCASE_SCHEMA: String = "PARQUET_LOWERCASE_SCHEMA"
 
+  /** An optional proxy host that needs to be used when accessing the bucket */
```
Comment (morazow) [inline src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala:154]:
```suggestion
  /** An optional proxy host that needs to be used when accessing the bucket. */
```

Comment (morazow) [reply src/main/scala/com/exasol/cloudetl/storage/StorageProperties.scala:154]: Change other also.

**Review by morazow — CHANGES_REQUESTED** [review]
I have added some minor editorial suggestions.

**Comment by baunz** [comment]
Hi @morazow 

thanks for reviewing and fixing - I did not had time yesteray to go over your suggestions but will do this today!

**Comment by morazow** [comment]
Hey @baunz,

Thanks a lot for the pull request! We are planning new release soon, it would be nice to have this also included.

**Pull request merged**